src/java.base/share/classes/com/sun/crypto/provider/AEADBufferedStream.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.crypto.provider;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.HexFormat;
+
+/**
+ * This class extends ByteArrayOutputStream by optimizing internal buffering.
+ * It skips bounds checking, as the buffers are known and input previously
+ * checked.  toByteArray() returns the internal buffer to avoid an extra copy.
+ *
+ * This uses `count` to determine the state of `buf`.  `buf` can still
+ * point to an array while `count` equals zero.
+ */
+final class AEADBufferedStream extends ByteArrayOutputStream {
+    /**
+     * Create an instance with no buffer
+     */
+    public AEADBufferedStream() {
+        buf = null;
+        count = 0;
+    }
+
+    /**
+     * Create an instance with the specified buffer
+     */
+
+    public AEADBufferedStream(int len) {
+        buf = new byte[len];
```

Comment:
```suggestion
        super(len);
```
otherwise buf will be initialized twice, once here, once in the base class constructor